### DISCUSSION TITLE
Should I use JBrowse 1 or 2 and advice on getting started.

### DISCUSSION THREAD

**Discussion by ghost** [GMOD/jbrowse-components#1570] [General]
I need help getting started with JBrowse. I have a background in website coding but NOT in molecular biology, which makes it difficult to understand what I need from JBrowse in order to know if I should go with JBrowse 1 or 2. My stakeholder has a background in Molecular Virology, Biology, and Immunology. 

Our concept website will display multiple HIV-1 sequences (aligned to HXB2) and go on to present analysis of user selected HIV-1 strains resistance to bNAbs. It will identify patterns of resistance and the mechanisms that allow HIV-1 strains to escape neutralization.

Is JBrowse 2 a platform I need given our focus on HIV-1? Will JBrowse 1 display multiple sequences in tracks?

If the answer is JBrowse 2, should I start with a docker image in order to demo JBrowse 2 to my stake holder, or should I put on my developer's cap and use the instructions for JBrowse 2 development? My demo needs to show multiple HIV-1 sequences aligned to HXB2. 

Thanks!

**Comment by garrettjstevens** [comment]
Hi, thanks for your interest in JBrowse! Can I get some clarification of what you mean by "show multiple HIV-1 sequences aligned to HXB2"? What is the format of the data you're wanting to visualize? For example, are you wanting to visualize the results of a whole-sequence alignment of a sequence to HXB2  (maybe in PAF format)? Or are you wanting to visualize genome sequencing reads aligned to HXB2 (perhaps in BAM or CRAM format)?

**Comment by ghost** [comment]
Lol. Good questions.  HXB2 was selected as a prototype, because this virus is the most commonly used reference strain for many different kinds of functional studies. All other HIV-1 envelope structural data published to date translates residue numbers into the HXB2 numbering scheme.

I will need to come up with a strategy for loading all the existing HIV-1 strains into our application. I know the web site needs a sequence viewer and JBrowse seems the best open source to use. But I don't know yet how to load data from https://www.hiv.lanl.gov/content/index. 

I was recently focused on Chado and Tripal on Drupal (the university supports Drupal sites) but I've put that off until Tripal is on Drupal 9 and hopefully can run on other databases than Postgresql, only because the university doesn't run Postgresql on their Drupal servers. I do not have a commit from university resources on how to host this website if it is not a Drupal site. So I put all that aside and turned to JBrowse.

I should mention that I will eventually be writing code for this website. Currently I have WSL2/Ubuntu/Docker/Docker-Compose for local development setup. And I need to proceed with localhost development for at least the next 4 months. At that point I hope to understand how we like JBrowse, how progress is going on Tripal, and whether or not we want to move towards Drupal or stay as a nodeJS/JBrowse website.

Thanks!

**Comment by garrettjstevens** [comment]
Unless you have a specific reason to use JBrowse 1, I would suggest using JBrowse 2. We don't have a docker image, but I'd suggest starting with our [quick start guide](https://jbrowse.org/jb2/docs/quickstart_web). You can also browse our [demo instance](https://s3.amazonaws.com/jbrowse.org/code/jb2/v1.0.2/index.html) to make sure JBrowse 2 provides the features you are looking for.

**Comment by ghost** [comment]
Yes, I’ve been through the quick start guide for JBrowse 2. I think that I
should clone into my development environment, but before I do that I just
need to make sure that I shouldn’t stay with JBrowse 1 (I don’t understand
the feature set between the two) and that there isn’t a docker image to use
rather than cloning the repository.

Thanks for your response!

**Comment by scottcain** [comment]
I am definitely not disagreeing with Garrett, but using JBrowse 1 is not a
crazy idea, depending on how you want to display your data.  For instance
if you have variants in VCF and multiple sequence alignments, those are
well supported in JBrowse 1.  On the other hand, if you were looking to do
synteny type displays, JBrowse 2 is definitely the way you'd want to go.
For a JBrowse 1 example, the SARS-CoV-2 browser at covid19.jbrowse.org is
good:

http://covid19.jbrowse.org/jbrowse/?data=data%2FSARS-CoV-2&loc=NC_045512.2%3A17338..17441&tracks=SRR11140744%2CSRR11140746%2CAll%20Genes%2CVertebrate%20coronaviruses%2CMultiAlignment%20of%20GenBank%20seqs%20(slow)&highlight=

It has variant data from individual samples along with a multiple sequence
alignment that is a little slow to display, but HIV is only 1/3 the size of
SARS-CoV-2, so it might be better (or might not, depending on how many
sequences you put in there).  The server is driven by this docker image:

  https://hub.docker.com/r/gmod/sars-cov-2-jbrowse

Honestly, the docker container is pretty simple--it's writing the configs
for JBrowse that drive the display that is a little bit of work.  One of
the things I like about docker for JBrowse servers is that they are very
lightweight (typically under 200MB) and run with very little resources.


On Wed, Dec 16, 2020 at 4:59 PM pst024 <notifications@github.com> wrote:

> Yes, I’ve been through the quick start guide for JBrowse 2. I think that I
> should clone into my development environment, but before I do that I just
> need to make sure that I shouldn’t stay with JBrowse 1 (I don’t understand
> the feature set between the two) and that there isn’t a docker image to use
> rather than cloning the repository.
>
> Thanks for your response!
>
>
>
> On Wed, Dec 16, 2020 at 6:53 PM Garrett Stevens <notifications@github.com>
> wrote:
>
> > Unless you have a specific reason to use JBrowse 1, I would suggest using
> > JBrowse 2. We don't have a docker image, but I'd suggest starting with
> our quick
> > start guide <https://jbrowse.org/jb2/docs/quickstart_web>. You can also
> > browse our demo instance
> > <https://s3.amazonaws.com/jbrowse.org/code/jb2/v1.0.2/index.html> to
> make
> > sure JBrowse 2 provides the features you are looking for.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/GMOD/jbrowse-components/discussions/1570#discussioncomment-217767
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ARKDBPPHSF5R2HVXC7XLEXDSVFJADANCNFSM4U6TUJUA
> >
> > .
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GMOD/jbrowse-components/discussions/1570#discussioncomment-217778>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACDCP5MP63D6HA7CPNUJMLSVFJW5ANCNFSM4U6TUJUA>
> .
>


-- 
------------------------------------------------------------------------
Scott Cain, Ph. D.                                   scott at scottcain dot
net
GMOD Coordinator (http://gmod.org/)                     216-392-3087
Ontario Institute for Cancer Research

**Comment by ghost** [comment]
Hi Scott,

I have been asking myself that since HIV-1 is only ~2600bp, maybe JBrowse 1 is the right path. We will be creating graphs about the data, but not synteny type displays of the data. 

The only reason that I don't know file formats for HIV-1 and how to config for JBrowse is that DevOps is overwhelming me at the present. JBrowse 2, which is what I started looking at, uses Yarn. Should I stay on that path or use npm with JBrowse 1. These are the kind of questions I'm grappling with. 

Thanks!

**Comment by rbuels** [comment]
One other angle, if you are embedding this inside another app (which it sounds like you are), JB2 has much more advanced support for embedding as a React component or using plain JS.

**Comment by ghost** [comment]
Ahh. Thanks for this advice. In a similar vein, are all the current
JBrowse1 plugins available (or soon to be) for JBrowse2? Playing with the
sars-covid-2-jbrowse, I believe it is built on JBrowse 1 and uses plugins.
That is what I see in the Dockerfile.

**Reply by rbuels** [reply]
Well, JB1 plugins aren't runnable in JB2, because the code is very different. My hope is that over time many JB1 plugins will be ported by their authors to run in JB2, but that is up to the individual plugin authors.
